git_machete/github.py

```diff
@@ -76,6 +76,12 @@ class GithubTokenAndTokenProvider(NamedTuple):
     token_provider: str
 
 
+class GitHubClient(NamedTuple):
```

Comment:
Whoops that's not a service client class... I'd rather expect the functions like `create_pull_request` to become its methods!

Reply:
(or in case this is WIP, pls mark the PR as draft)